tests/test_infer.py

```diff
@@ -160,12 +161,12 @@ def test_calculator_distributions_without_teststatistic(qtilde):
 
 
 @pytest.mark.parametrize(
-    "backend",
+    "backend_rtol",
```

Comment:
```suggestion
    "backend,rtol",
```

Reply:
sorry, forgot to mention that - i tried to do it like that before, but it did not work. I get the following error:
```
=========================================================================================== ERRORS ============================================================================================
____________________________________________________________________________ ERROR collecting tests/test_infer.py _____________________________________________________________________________
../../venv/pyhf/lib/python3.8/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
../../venv/pyhf/lib/python3.8/site-packages/pluggy/manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../venv/pyhf/lib/python3.8/site-packages/pluggy/manager.py:84: in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
../../venv/pyhf/lib/python3.8/site-packages/_pytest/python.py:224: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
../../venv/pyhf/lib/python3.8/site-packages/_pytest/python.py:409: in _genfunctions
    self.ihook.pytest_generate_tests(metafunc=metafunc)
../../venv/pyhf/lib/python3.8/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
../../venv/pyhf/lib/python3.8/site-packages/pluggy/manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../venv/pyhf/lib/python3.8/site-packages/pluggy/manager.py:84: in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
../../venv/pyhf/lib/python3.8/site-packages/_pytest/python.py:133: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
../../venv/pyhf/lib/python3.8/site-packages/_pytest/python.py:966: in parametrize
    newcallspec.setmulti2(
../../venv/pyhf/lib/python3.8/site-packages/_pytest/python.py:841: in setmulti2
    self._checkargnotcontained(arg)
../../venv/pyhf/lib/python3.8/site-packages/_pytest/python.py:825: in _checkargnotcontained
    raise ValueError("duplicate %r" % (arg,))
E   ValueError: duplicate 'backend'
```

For the other test, where i don't have these "stacked" decorators it seems to work

Reply:
ah - i think i have it now. The order matters - need to apply the two decorators in different order

Reply:
unfortunately i don't really know why or where to look it up :wink:

Reply:
call it something other than `backend` and you're fine. the problem is we have a global fixture called `backend` that allows us to run a test automatically across all backends by specifying backend as a parameter.

Reply:
oh - i think then it is maybe better to use that fixture here as well. I did that now - switching the tolerances based on the set dtypes of torch or tensorflow. However, the tests for pytorch64 only pass with the fix from #906.